core/src/main/scala/cats/std/string.scala

```diff
@@ -1,7 +1,24 @@
 package cats
 package std
 
-trait StringInstances extends algebra.std.StringInstances {
+trait StringInstances {
+  implicit val stringOrder: Order[String] = new StringOrder
+  implicit val stringMonoid = new StringMonoid
   implicit val stringShow: Show[String] =
     Show.fromToString[String]
 }
+
+class StringOrder extends Order[String] {
```

Comment:
Oh good point. I had in my head that all of the changes in this PR were part of the kernel module, but now I see the flaw in that reasoning :)

I still would advocate the suggested changes to be consistent with the rest of cats, and I think binary compatibility is still a worthwhile goal outside of kernel.